sumcheck/src/prover/plain.rs

```diff
@@ -0,0 +1,182 @@
+
+                p0_evx += p0_delta;
+                p1_evx += p1_delta;
+                q0_evx += q0_delta;
+                q1_evx += q1_delta;
+                eq_evx += eq_delta;
+                let eval_point_3 = comb_func(p0_evx, p1_evx, q0_evx, q1_evx, eq_evx, r_batch);
+
+                (eval_point_1 + a, eval_point_2 + b, eval_point_3 + c)
+            });
+
+        #[cfg(feature = "concurrent")]
+        let (eval_point_1, eval_point_2, eval_point_3) = (0..len)
+            .into_par_iter()
+            .fold(
+                || (E::ZERO, E::ZERO, E::ZERO),
+                |(a, b, c), i| {
+                    let eval_point_1 = comb_func(
+                        p0[2 * i + 1],
+                        p1[2 * i + 1],
+                        q0[2 * i + 1],
+                        q1[2 * i + 1],
+                        eq[2 * i + 1],
+                        r_batch,
+                    );
+
+                    let p0_delta = p0[2 * i + 1] - p0[2 * i];
+                    let p1_delta = p1[2 * i + 1] - p1[2 * i];
+                    let q0_delta = q0[2 * i + 1] - q0[2 * i];
+                    let q1_delta = q1[2 * i + 1] - q1[2 * i];
+                    let eq_delta = eq[2 * i + 1] - eq[2 * i];
+
+                    let mut p0_evx = p0[2 * i + 1] + p0_delta;
+                    let mut p1_evx = p1[2 * i + 1] + p1_delta;
+                    let mut q0_evx = q0[2 * i + 1] + q0_delta;
+                    let mut q1_evx = q1[2 * i + 1] + q1_delta;
+                    let mut eq_evx = eq[2 * i + 1] + eq_delta;
+                    let eval_point_2 = comb_func(p0_evx, p1_evx, q0_evx, q1_evx, eq_evx, r_batch);
+
+                    p0_evx += p0_delta;
+                    p1_evx += p1_delta;
+                    q0_evx += q0_delta;
+                    q1_evx += q1_delta;
+                    eq_evx += eq_delta;
+                    let eval_point_3 = comb_func(p0_evx, p1_evx, q0_evx, q1_evx, eq_evx, r_batch);
+
+                    (eval_point_1 + a, eval_point_2 + b, eval_point_3 + c)
+                },
+            )
+            .reduce(
+                || (E::ZERO, E::ZERO, E::ZERO),
+                |(a0, b0, c0), (a1, b1, c1)| (a0 + a1, b0 + b1, c0 + c1),
+            );
+
+        let evals = smallvec![eval_point_1, eval_point_2, eval_point_3,];
+        let poly = CompressedUnivariatePolyEvals(evals);
+        let round_poly_coefs = poly.to_poly(claim);
+
+        // reseed with the s_i polynomial
+        transcript.reseed(H::hash_elements(&round_poly_coefs.0));
+        let round_proof = RoundProof {
+            round_poly_coefs: round_poly_coefs.clone(),
+        };
+
+        round_proofs.push(round_proof);
+
+        let round_challenge =
+            transcript.draw().map_err(|_| SumCheckProverError::FailedToGenerateChallenge)?;
+
+        // compute the new reduced round claim
+        let new_claim = round_poly_coefs.evaluate_using_claim(&claim, &round_challenge);
```

Comment:
this statement can be moved after the calls to `bind_least_significant_variable()` (and assigned to `claim` directly)